lib/Views/render.jsx

```diff
@@ -1,21 +1,48 @@
+import { observer } from "mobx-react";
+import PropTypes from "prop-types";
+import React, { Suspense } from "react";
 import { createRoot } from "react-dom/client";
-import { flushSync } from "react-dom";
 import Variables from "../Styles/variables.scss";
-import UI from "./UserInterface";
+import "./global.scss";
+import { Loader } from "./Loader";
+import { terriaStore } from "./terriaStore";
 
-export default function renderUi(terria, allBaseMaps, viewState) {
-  const container = document.getElementById("ui");
-  const root = createRoot(container);
+// Lazy load the entire TerriaUserInterface component
+const LazyTerriaUserInterface = React.lazy(() =>
```

Comment:
UI needs to be lazy-loaded, or Webpack won't separate the code into chunks, leading to broken loader

Reply:
That information is good to have in the comment.